### ISSUE TITLE
Some domain test failures

### ISSUE THREAD

**Issue by brodybits** [jxcore/jxcore#529]
I notice that the following tests fail, at least on my Mac Air system:
- test/simple/test-crypto-domains.js
 -test/simple/test-domain-top-level-error-handler-throw.js
- test/simple/test-domain-with-abort-on-uncaught-exception.js

I spent a few hours looking at these tests but am still pretty clueless. I wonder if timing could be a factor, especially for test-domain-top-level-error-handler-throw.js. If there is an easy way to fix these by emulate timing of node.js or something, it would be an even stronger proof of compatibility with node.js.

If we want to disable or remove these tests, I am fine with it as well.

**Comment by ktrzeciaknubisa** [comment]
Yes, I saw it too: https://github.com/jxcore/jxcore/issues/229, https://github.com/jxcore/jxcore/issues/228. I guess it still awaits to be resolved :)

**Comment by obastemur** [comment]
@brodybits @ktrzeciaknubisa I didn't bring the domain related couple of changes from the core. So far keeping these tests to remind what is different than node.js. I'm open to suggestions though.

**Comment by brodybits** [comment]
Agreed. My only remark then is to document this somewhere in README.md or TESTING.md.

Also these failures block the jslint stage. If I run `make jslint` I get a huge number of errors. I already fixed a bunch in `lib/jx` (in my own workarea), not sure when I will have time to fix the rest.

**Comment by obastemur** [comment]
>  I already fixed a bunch in lib/jx (in my own workarea), not sure when I will have time to fix the rest.

PR on that is really appreciated.

**Comment by brodybits** [comment]
> PR on that is really appreciated.

Coming soon!

**Comment by brodybits** [comment]
> PR on that is really appreciated.

Just issued PR #531 to fix the jslint errors in `lib/jx`.

**Comment by brodybits** [comment]
@ktrzeciaknubisa @obastemur what is `lib/jx/_jx_marker.js` and why do we need it?

I wanted to exclude `lib/jx/_jx_marker.js` from jslint but could not figure out how to exclude multiple file paths in the gjslint command.

**Comment by obastemur** [comment]
That's the marker file for native packaging. I can move that into native code.

**Comment by brodybits** [comment]
> That's the marker file for native packaging. I can move that into native code.

That would be great. Can we document this (and how the native packaging works in general) somewhere?

**Comment by brodybits** [comment]
I pushed the remaining jslint fixes in #541. I recommend that we ask _all_ contributors to test their own work using `make test` before submitting. Also we should ask all contributors to run `make jslint` since it is blocked by the failing tests.

I hope to look at the missing domain functionality later this weekend, no promise though.

**Comment by obastemur** [comment]
> I hope to look at the missing domain functionality later this weekend, no promise though.

I don't know if anyone is using that though. We may just skip them and consider as a difference of JXcore (one of them requires a tweak on V8) That's the reason I didn't bring it at the first place. There are some cases latest revisions of Node 10 fails to run some node apps while jxcore is able to run as is. I'm going to move those tests into a separate folder and document that. Any Objection ?

**Comment by ktrzeciaknubisa** [comment]
> I'm going to move those tests into a separate folder and document that. Any Objection ?

I'm not sure about making exceptions. Even small node's feature exclusion may come back to us later with double impact/double pain. 

> I don't know if anyone is using that though.

Yeap, we don't know. But even if users do not use that directly, some node modules plugged into complex projects may use it.

My opinion is, that we should stick to keeping maximum of node features compatibility, unless there is really good reason for this.

**Comment by brodybits** [comment]
I would agree with @ktrzeciaknubisa, but I think it would be valid to ask the question _when_ we want to do this. I think we _know_ nobody is actually depending on the broken functionality right now. For example, we _could_ say that we add the Node 4.0 functionality first, hopefully with the V8 tweaks that we need working, and then fix this issue on the newer release.

A side remark is that io.js/Node 4.0 moved the tests around anyway, so I don't see much harm in moving them into another folder until someone gets a chance to fix it.

**Comment by obastemur** [comment]
Keeping this one open while the failing tests moved into `test/TODO-0.3.x` folder.

**Comment by brodybits** [comment]
:+1:
